tlsredis.go

```diff
@@ -137,17 +135,14 @@ func GetClient(opts *Options) (*redis.Client, error) {
 		}
 	}
 
-	opt := redis.Options{
-		Dialer:   dialFunc,
-		DB:       db,
-		PoolSize: opts.PoolSize,
-	}
+	opts.Dialer = dialFunc
+	opts.DB = db
 	if u.User != nil {
 		redisPass, _ := u.User.Password()
-		opt.Password = redisPass
+		opts.Password = redisPass
```

Comment:
Sorry, for some reason I misread the signature of this function.  I realize clients of this function would really need to go out of their way to shoot themselves in the foot to trigger this (mutate and reuse Options objects), so this may be OK by convention.